doc/shared_library_howto/howto.txt

```diff
@@ -0,0 +1,107 @@
+
+Compiling and Using Theano functions as shared libraries
+========================================================
+
+First one important note: this functionality is experimental, so not every
+features works, and some may not work as you expected.
+
+This functionality was only be tested on CPU code on a Linux machine.
+
+For now, shared variables not works.
+
+Clearly every node of function that you want to use as shared library must
+have a C implementation.
+ 
+To compile a function as shared library, you must use the CLinker with the the
+parameter c_callable=True:
+
+linker = theano.gof.CLinker(c_callable=True)
+mode = theano.Mode(linker=linker)
+f = theano.function(inputs, outputs , mode=mode)
+
+The shared library is build in a specific temp directory, in the Theano cache
+directory. To know in which directory, you can call print the fn.name
+property of the function
+
+print f.fn.filename
+
+In this directory you find the .so library, the header file, a makefile and a
+exe.cpp which can be used as template to build your application. There is also
+a exec.h in which you find the command used to build the library.
+
+Let's walk through an example to see how use the shared library.
+
+Build the shared library for the function in function.py:
+
+python function.py
+
+the outputs show you where Theano puts the library. To see if everything
+works correctly go in that directory and launch the exec executable. You must
+get:
+
+After import numpy (hexvalue)
+after import_array1()
+after cinit()
+run() from the shared library returned=0
+[[  0.   2.   4.   6.]
+ [  8.  10.  12.  14.]
+ [ 16.  18.  20.  22.]]
+[ 20.  23.  26.  29.]
+main end, before Py_Finalize
+
+Please note that, in general case, the exec can give some errors for generic
+functions because it makes some assumption on the data the can be wrong. Take
+it as a simple template. 
+
+Theano compiles the function as a C++ class with variables mapped to internal
+variables. To run the function you initialized the class calling the cinit
+function. Then pass the values of inputs variables as numpy array and call the
+run method. If there is no errors you can get the outputs values from C++
+class.
+To C++ class variable must be set and get using the PyList_SetItem and
```

Comment:
I have no idea what this sentence is supposed to be, but it is wrong (syntactically).